framework/db/mysql/Schema.php

```diff
@@ -293,9 +293,9 @@ protected function loadColumnSchema($info)
              *
              * See details here: https://mariadb.com/kb/en/library/now/#description
              */
-            if (($column->type === 'timestamp' || $column->type === 'datetime') &&
-                ($info['default'] === 'CURRENT_TIMESTAMP' || $info['default'] === 'current_timestamp()')) {
-                $column->defaultValue = new Expression('CURRENT_TIMESTAMP');
+            if (($column->type === 'timestamp' || $column->type === 'datetime')
+                && preg_match('/^current_timestamp(?:\(([0-9]*)\))?$/i', $info['default'], $matches)) {
+                $column->defaultValue = new Expression('CURRENT_TIMESTAMP' . (!empty($matches[1]) ? '(' . $matches[1] . ')' : ''));
```

Comment:
You can add full `current_timestamp(1)` to a capturing group to simplify the second expression. It will be a bit more readable and maintainable.

Reply:
Are you sure that `new Expression(strtoupper($matches[1]))` is more readable?

Reply:
Current one is OK.